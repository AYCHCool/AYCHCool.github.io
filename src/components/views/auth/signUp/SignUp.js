import React, { useState, useEffect, Fragment } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import Toastr from 'toastr';
import { getToken, authorizeSocialUser } from '@Redux/actions/socialActions';
import { registerAccount } from '@Redux/actions/authActions';
import InputField from '@Common/form/InputField';
import LeftDiv from '@Common/auth/leftDiv';
import SocialButtons from '@Common/auth/buttons';
import { signupText } from '@Common/auth/leftDivText';
import '@Common/auth/auth.scss';

const SignUp = (props) => {
  const [values, setValues] = useState({
    firstName: '',
    lastName: '',
    username: '',
    email: '',
    password: '',
    confirmPassword: '',
  });
  useEffect(() => {
    const { location, socialSignOn, history } = props;
    const tokenString = location && location.search;
    if (tokenString) {
      const token = getToken(tokenString);
      socialSignOn(token);
      history.push('/');
      Toastr.success('Authentication was successful!');
    }
  }, [props]);

  const { loading, history, onSubmit } = props;
  const onChange = (event) => {
    event.persist();
    setValues(prevState => ({ ...prevState, [event.target.name]: event.target.value }));
  };
  const registerUser = (event) => {
    event.preventDefault();
    if (values.password !== values.confirmPassword) {
      Toastr.error('Passwords do not match');
      return;
    }
    onSubmit(values, history);
  };
  return (
    <Fragment>
      <div className="container-body">
        <LeftDiv text={signupText} />
        <div className="right-block">
          <form
            onSubmit={registerUser}
            className="main-form-signup"
          >
            <div className="flex-input">
              <div className="flex-field">
                <div className="input-field">
                  <InputField
                    type="text"
                    value={values.firstName}
                    onChange={onChange}
                    name="firstName"
                    placeholder="First Name"
                    className="form-input"
                    required
                  />
                </div>
              </div>
              <div className="flex-field">
                <div className="input-field">
                  <InputField
                    type="text"
                    value={values.lastName}
                    onChange={onChange}
                    name="lastName"
                    placeholder="Last Name"
                    className="form-input"
                    required
                  />
                </div>
              </div>
            </div>
            <div className="input-field">
              <InputField
                type="email"
                value={values.email}
                onChange={onChange}
                name="email"
                placeholder="Email"
                className="form-input"
                required
              />
            </div>
            <div className="input-field">
              <InputField
                type="password"
                value={values.password}
                onChange={onChange}
                name="password"
                placeholder="Password"
                className="form-input"
                required
              />
            </div>
            <div className="input-field">
              <InputField
                type="password"
                value={values.confirmPassword}
                onChange={onChange}
                name="confirmPassword"
                placeholder="Confirm Password"
                className="form-input"
                required
              />
            </div>
            <div className="input-field">
              <button className="btn--block bttn-primary btn-submit" type="submit" value="Sign up" disabled={loading}> {
                loading ? 'Loading...' : 'Create your free account'
              }
                <i className="arrow-forward material-icons">arrow_forward</i>
              </button>
            </div>
          </form>
          <div className="input-field">
            <h2 id="htwo-space">
              <div className="input-row">
                <div className="column1"><hr /></div>
                <div className="column2">OR</div>
                <div className="column1"><hr /></div>
              </div>
            </h2>
          </div>
          <SocialButtons />
        </div>
      </div>
    </Fragment>
  );
};
SignUp.propTypes = {
  onSubmit: PropTypes.func.isRequired,
  errors: PropTypes.shape({}),
  auth: PropTypes.shape({}),
  social: PropTypes.shape({}),
  history: PropTypes.shape({}).isRequired,
  loading: PropTypes.bool.isRequired,
};
SignUp.defaultProps = {
  errors: {},
  auth: {},
  social: {},
};

const mapStateToProps = state => ({
  auth: state.auth,
  errors: state.auth.errors,
  loading: state.auth.loading,
  social: state.social,
});
const mapDispatchToProps = dispatch => ({
  onSubmit: (newUser, history) => dispatch(registerAccount(newUser, history)),
  socialSignOn: token => dispatch(authorizeSocialUser(token)),
});

// export const onRegisterUser = (newUser, history) => registerAccount(newUser, history);
export const SignUpComponent = SignUp;

export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(SignUp);
