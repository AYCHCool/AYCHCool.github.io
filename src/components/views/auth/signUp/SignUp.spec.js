import React from 'react';
import { shallow, mount } from 'enzyme';
import { BrowserRouter as Router } from 'react-router-dom';
import store from '@Redux/store/index';
import InputField from '@Common/form/InputField';
import SignUp, { SignUpComponent } from './SignUp';

const props = {
  loading: false,
  history: {
    push: jest.fn(),
  },
};

describe('SignUp Component', () => {
  it('Should render without errors', () => {
    const component = shallow(
      <Router>
        <SignUp store={store} {...props} />
      </Router>,
    );
    expect(component).toMatchSnapshot();
  });
  it('Should render along with children component', () => {
    const component = mount(
      <Router>
        <SignUp store={store} {...props} />
      </Router>,
    );
    expect(component).toMatchSnapshot();
    expect(component.find('button')).toHaveLength(3);
    expect(component.find('input')).toHaveLength(5);
    expect(component.find('img')).toHaveLength(1);
  });

  it('should call onChange prop with input value', () => {
    const onChange = jest.fn();
    const component = mount(<InputField value="custom email" type="email" name="email" onChange={onChange} />);
    expect(component).toMatchSnapshot();
  });

  it('email Input check', () => {
    const onChange = jest.fn();
    const registerUser = jest.fn();
    const event = {
      persist: jest.fn(),
      target: { value: 'james@email.com', name: 'email' },
    };
    const wrapper = shallow(<SignUpComponent onSubmit={registerUser} onChange={onChange} {...props} />);
    wrapper.find('Input[type="email"]').at(0).simulate('change', event);
    expect(wrapper.find('input[name="email"]')).toMatchSnapshot();
  });
  it('Password Input check', () => {
    const onChange = jest.fn();
    const registerUser = jest.fn();
    const event = {
      persist: jest.fn(),
      target: { value: 'P455word', name: 'password' },
    };
    const wrapper = shallow(<SignUpComponent onSubmit={registerUser} onChange={onChange} {...props} />);
    wrapper.find('Input[type="password"]').at(0).simulate('change', event);
    expect(wrapper.find('input[name="password"]')).toMatchSnapshot();
  });

  it(' should call onSubmit prop function when form is submitted', () => {
    const registerUser = jest.fn();
    const wrapper = mount(<SignUpComponent onSubmit={registerUser} {...props} />);
    const form = wrapper.find('form');
    form.simulate('submit');
    expect(registerUser).toHaveBeenCalledTimes(1);
  });
});
