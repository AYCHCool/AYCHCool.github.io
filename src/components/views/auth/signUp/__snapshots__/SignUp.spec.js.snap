// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignUp Component Password Input check 1`] = `ShallowWrapper {}`;

exports[`SignUp Component Should render along with children component 1`] = `ReactWrapper {}`;

exports[`SignUp Component Should render without errors 1`] = `ShallowWrapper {}`;

exports[`SignUp Component email Input check 1`] = `ShallowWrapper {}`;

exports[`SignUp Component should call onChange prop with input value 1`] = `ReactWrapper {}`;
